Sources/Kommunicate/Classes/KMConversationService.swift

```diff
@@ -93,17 +93,27 @@ public class KMConversationService: KMConservationServiceable,Localizable {
                                 case .success:
                                     dispatchGroup.leave()
                                 case .failure(let error):
-                                    let errorResponse = Response(success: false, clientChannelKey: clientId, error: error)
-                                    completion(errorResponse)
+                                    response.error = error
+                                    response.success = false
                                     dispatchGroup.leave()
                                 }
                             }
                         }
                     }
                     dispatchGroup.enter()
-                    guard self.groupMetadata != nil else { return }
-                    self.updateGroupMetadata(groupId: NSNumber(value: groupID), channelKey: "", metadata: self.groupMetadata) { response in
+                    guard self.groupMetadata != nil else {
                         dispatchGroup.leave()
+                        return
```

Comment:
We've a return statement here, so just check if it will still go to ` dispatchGroup.notify` block.